### PR TITLE
Fix reducing similar action names

### PR DESCRIPTION
**Pull request by kinetifex**
Fix edge case where actions for similar API names we being reduced improperly (i.e. `api_START` and `api2_START`).

### CODE REVIEW

File: tests/reducers.spec.js [hunk 1]
```diff
@@ -128,6 +128,13 @@ describe('Reducers', () => {
       expect(handlers.onComplete).not.toHaveBeenCalled();
       expect(handlers.onReset).toHaveBeenCalled();
     });
+
+    it('does not handler for similarly named apis', () => {
```
Comment (rmarkins-godaddy) [inline tests/reducers.spec.js:132]: typo 
```suggestion
    it('does not handle for similarly named apis', () => {
```

File: src/reducers.js [hunk 2]
```diff
@@ -59,11 +57,14 @@ handlers.onComplete = function (state, action) {
  * @private
  */
 export default function createReducer(apiName) {
+  const reApiAction = new RegExp(`^${apiName}_(RESET|START|SUCCESS|FAIL)$`);
```
Comment (DullReferenceException) [inline src/reducers.js:60]: Could `apiName` contain non-regex-safe characters? Might want to do some escape sequences. This might be a good guide for that: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping

Comment (kinetifex) [reply src/reducers.js]: Good point, better safe than sorry. Fixed in fbac4d7
